File: api/src/main/java/club/yunzhi/smartcommunity/service/UserDataStatisticsService.java
package club.yunzhi.smartcommunity.service;

import club.yunzhi.smartcommunity.entity.WebUserDataStatistics;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;


/**
 * 用户数据统计
 */
public interface UserDataStatisticsService {
  /**
   * 获取某个区域下管理的所有的用户信息
   * @param districtId 区域ID
   * @param pageable 分页
   */
  Page<WebUserDataStatistics> pageByBelongDistrictId(Long districtId, Pageable pageable);

  /**
   * 重新生成数据
   */
  void reGenerateData();
}

File: web/projects/lib/src/service/user-data-statistics.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {Observable} from 'rxjs';
import {WebUserDataStatistics} from '../entity/web-user-data-statistics';
import {Page} from '@yunzhi/ng-common';

@Injectable({
  providedIn: 'root'
})
export class UserDataStatisticsService {
  private readonly baseUrl = 'UserDataStatistics';

  constructor(private httpClient: HttpClient) {
  }

  pageTop10BelongDistrictId(districtId: number): Observable<Page<WebUserDataStatistics>> {
    const httpParams = new HttpParams()
      .append('page', 0)
      .append('size', 10)
      .append('sort', 'enterCount,desc');
    return this.httpClient.get<Page<WebUserDataStatistics>>(this.baseUrl + '/pageByBelongDistrictId/' + districtId,
      {params: httpParams});
  }
}
